src/components/SectionCampaign/SectionCampaign.ts

```diff
@@ -49,7 +49,8 @@ export class SectionCampaign extends NostoElement {
 
 async function getSectionMarkup(element: SectionCampaign, rec: JSONResult) {
   const handles = rec.products.map(product => product.handle).join(":")
-  const target = new URL("/search", window.location.href)
+  const root = window.Shopify?.routes?.root ?? "/"
+  const target = new URL(`${root}search`, window.location.href)
```

Comment:
Why do we have different treatment for URL generation in DynamicCard (line 89) and this logic?
If they are same, we can maybe move the Shopify.routes logic into `fetchWithErrorHandling`

Reply:
Okay, maybe move it to a reusable logic that generates the URL?